third_party/externals.bzl

```diff
@@ -3,6 +3,13 @@ load("//third_party/cargo:crates.bzl", "raze_fetch_remote_crates")
 
 # We define our externals here instead of directly in WORKSPACE
 def register_sorbet_dependencies():
+    http_archive(
+        name = "platforms",
+        urls = _github_public_urls("bazelbuild/platforms/archive/d4c9d7f51a7c403814b60f66d20eeb425fbaaacb.zip"),
```

Comment:
bazel 4.1 has decided to split definitions of platforms out into a separate tarball?  o.O

(I mean, I think I can understand why you would want to do that, but still seems a little weird.)

Reply:
This is definitely weird, and is caused by something in the sorbet repository (making a fresh empty bazel project that uses bazel-4.1 doesn't have the problem with macos builds that pinning the platforms package fixes). I think it would be great if we didn't need this, but one of our dependencies is messing with it :(

Reply:
OK, reading deeper I found a link to this issue: https://github.com/bazelbuild/bazel/issues/8622. Apparently the future is to depend on the platforms repository directly if you need it, and that if you rely on `@platforms` without it you'll get whatever snapshot happens to be included with that release of bazel. I think it might be worth putting `--incompatible_use_platforms_repo_for_constraints` in our .bazelrc to make this explicit.

Reply:
Turns out we can't turn on `--incompatible_use_platforms_repo_for_constraints` yet as it breaks the version of abseil that we're using. At some point it would be great to iterate on updating dependencies so to the point that we can turn that flag on in .bazelrc.